File: weekend-core/src/main/java/com/github/aly8246/core/util/MongoTemplateUtil.java
package com.github.aly8246.core.util;

import org.springframework.data.mongodb.core.MongoTemplate;
import org.springframework.stereotype.Component;


@Component
public class MongoTemplateUtil {
private MongoTemplate mongoTemplate2;

@javax.annotation.Resource
public void setMongoTemplate2(MongoTemplate mongoTemplate) {
	this.mongoTemplate2 = mongoTemplate;
	MongoTemplateUtil.mongoTemplate = mongoTemplate;
}

public static MongoTemplate mongoTemplate;
}

File: weekend-core/src/main/java/com/github/aly8246/core/util/MongoTemplateUtil.kt
package com.github.aly8246.core.util

import org.springframework.data.mongodb.core.MongoTemplate
import org.springframework.stereotype.Component


@Component
class MongoTemplateUtil {
    private var mongoTemplate2: MongoTemplate? = null

    @javax.annotation.Resource
    fun setMongoTemplate2(mongoTemplate: MongoTemplate) {
        this.mongoTemplate2 = mongoTemplate
        MongoTemplateUtil.mongoTemplate = mongoTemplate
    }

    companion object {
        lateinit var mongoTemplate: MongoTemplate
    }
}
